frontends/ast/simplify.cc

```diff
@@ -1093,7 +1186,9 @@ bool AstNode::simplify(bool const_fold, bool in_lvalue, int stage, int width_hin
 
 	// in certain cases a function must be evaluated constant. this is what in_param controls.
 	if (type == AST_PARAMETER || type == AST_LOCALPARAM || type == AST_DEFPARAM || type == AST_PARASET || type == AST_PREFIX)
-		in_param = true;
+		in_param_ = true;
+	log_assert(in_param == in_param_);
+	log_assert(in_lvalue == in_lvalue_);
```

Comment:
I want to point out these asserts (comparing the flags on the node with the flags passed in to simplify) are upheld through `make test`.